### ISSUE TITLE
Support for custom File Systems of VSCode

### ISSUE THREAD

**Issue by gbrueckl** [RandomFractals/vscode-data-preview#309]
It seems that the data preview currently does not support custom file systems created for VSCode using the (https://code.visualstudio.com/api/extension-guides/virtual-documents, https://github.com/microsoft/vscode-extension-samples/blob/main/fsprovider-sample/README.md)

So I have a custom file system and a file stored on it (e.g. `foo://myfolder/myfile.csv`). Everything works fine in VSCode itself (opening it from explorer, previewing, etc.) but the extensions does not seem to be able to load that file complaining about that `/myfolder/myfile.csv` does not exists

Is this supposed to work?
if not, any plans to add this feature?

thanks in advance, 
-gerhard

**Comment by RandomFractals** [comment]
That is correct. Data preview predates `workspace.fs` and newer apis vscode team added later for reading files in vscode and webview extensions. 

Could you provide a specific use case where you use CSV data files with a custom file scheme?

For the time being only local data files and remote files served over `https:/` are supported in data preview.

Also, worth noting that data preview dependencies are very outdated, and will be updated soon either in this extension or the new tabular data viewer I am working on.

**Comment by RandomFractals** [comment]
Sure, those are good use cases.

I am still debating if they will be supported in data preview, or the new tabular data viewer I'll be working on in the remaining months this year.

Relevant tickets in that repo and alpha v. I just made public again:

https://github.com/RandomFractals/tabular-data-viewer/issues/114

https://github.com/RandomFractals/tabular-data-viewer/issues/46

More info about that under our TablularDataViewer hashtag on twitter ... 

https://twitter.com/hashtag/TabularDataViewer?src=hashtag_click

**Comment by gbrueckl** [comment]
So will tabular-data-viewer be a successor of data-preview?
In that case it would be fine to have it there only I think

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by RandomFractals** [comment]
@gbrueckl I have not decided on that part yet, but if you want to try the latest version of perspective viewer with large csv data files for now, that's the extension you should try. 

I will be adding other Pro sponsors only features to that extension in the months to come. Data preview updates are on hold till I get better sense of how that new viewer is shaping up with embedded analytics db, etc.
